scripts/script.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+/* Delegate what happens when the DOM is ready.
+ */
+$(document).ready(function(){
+    var nav = new HeaderToNav();
+    nav.getElementArray('h2, h3').buildElements().insert('body');
+});
+
+/* Register event handlers.
+ */
+$('body').on('click', '#toggleButton', toggleSideNav);
+
+
+var HeaderToNav = function() {
+    this.elementsToNav = null;
+    this.addToDom = '';  // string representation of elements to add to DOM.
+}
+
+
+/* For each header, add the jQuery element to an array.
+ * Use all h2/h3 headers: 'h2' are categories, 'h3' are message items.
+ * Concern separated to handle future changes of category/message element types.
+ */
+HeaderToNav.prototype.getElementArray = function (itemsToUse) {
+    this.elementsToNav = $(itemsToUse);
+    return this;
+}
+
+
+/* For each header item in the array, build the DOM element(s) to set.
+ * 'h2' headers are categories used to group the messages.
+ * 'h3' headers are the message items.
+ * Some string functions are used for speed over typical jquery methods.
+ */
+HeaderToNav.prototype.buildElements = function() {
+    var that = this;  // reference 'this' scope for different scope below.
+    this.elementsToNav.each(function(){
+        var text = $(this).text();
+        var id = '#' + $(this).attr('id');
+
+        // 'h2' headers are section categories used to group the messages.
+        // 'h3' headers are the message items.
+        var catClass = $(this).is('h2') ? ' class="navCategory"' : '';
+        that.addToDom += '<li' + catClass + '><a href="' + id + '">' + text + 
+                         '</a></li>';
+    });
+
+    // wrap 'li' items in 'ul' and 'nav'.
+    this.addToDom = '<nav id="side"><span id="toggleButton">&#9776;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;PyTA Menu</span><ul id="show">' + 
```

Comment:
`&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;` is always a no-no. 
to achieve spacing, wrap in an inline element, like a new `span` and set the css styling...